packages/actor-rdf-resolve-quad-pattern-hypermedia/lib/ActorRdfResolveQuadPatternHypermedia.ts

```diff
@@ -54,7 +54,7 @@ export class ActorRdfResolveQuadPatternHypermedia extends ActorRdfResolveQuadPat
     }
     const metadata: {[id: string]: any} = await firstPageMetadata();
 
-    if (!metadata.searchForms || !metadata.searchForms.values.length) {
+    if (!metadata.next && (!metadata.searchForms || !metadata.searchForms.values.length)) {
```

Comment:
Also datasets with a search form and with a next page should trigger this actor, but probably they will have to trigger it from their code, as sometimes fetching the next page is not the right thing to do during the query processing.

@rubensworks What do you think about this? This means of course that the QPF actor would be able to overrule this actor and say it is better at handling things with a search form with an spo in it, and that inside the QPF actor, you would call this actor when you want to get the next page.

Reply:
Yeah, this check is definitely too strict, I'd remove the `metadata.next` check. With this change,  Comunica will always fail when for example iterating over TPF pages when it reaches the last page.

Actors down the line should indeed invoke the next page themselves when needed.

I'll have to look into this PR in more detail, but I'm quite sure no changes should be needed in this file.

Reply:
If I didn't add this to the check the next page actor was never invoked because there are no `searchForms` in the document. The next page actor makes up its own `searchForm` out of the `hydra:next` link while the `ActorRdfResolveHypermediaQpf` just returns the one that is present in the source. Maybe this check should not happen when it is possible that actors create their own `searchForms`?

Reply:
I would suggest removing this check completely then.